src/views/activity/Activity.vue

```diff
@@ -25,7 +25,7 @@ div
                @change="(periodLength) => setDate(date, periodLength)")
     div.p-1(v-if="periodLength === 'day'")
       input.form-control(id="date" type="date" :value="date" :max="today"
-                         @change="setDate($event.target.value)")
+                         @change="setDate($event.target.value, periodLength)")
```

Comment:
This was the real dealbreaker, we were not passing the periodLength to this function here.